.ci/test

```diff
@@ -5,10 +5,30 @@ set -e
 # For the test step concourse will set the following environment variables:
 # SOURCE_PATH - path to component repository root directory.
 
+# not all platforms support readlink -f
+# SOURCE_PATH must be set, and treat it as absolute path
+rlink=""
+set +e
+readlink -f /tmp >/dev/null 2>&1
```

Comment:
Shouldn't we just directly exit if `readlink` is not 'working as expected'? The only flow I see here is that `rlink` is `readlink` and there is no other path... 
Also, how does this match with the documentation in 
https://github.com/gardener/gardener/blob/master/docs/development/local_setup.md#mac-os-x-only-install-gnu-core-utilities ? This already explains that `readlink` on darwin does not work as expected.

Reply:
> Shouldn't we just directly exit if readlink is not 'working as expected'? 

Other than this, it works nicely on other systems, so why exit?

> The only flow I see here is that rlink is readlink and there is no other path

readlink without `-f` returns `""` if it is not a symlink, so not good.

> Also, how does this match with the documentation

I have some quite good and intelligently thought out answers to this question, but the honest one would be, "I plum forgot about that when I got to testing." Either way, `readlink -f` wouldn't work, only `greadlink`.

Reply:
> Other than this, it works nicely on other systems, so why exit?

You check at the top whether `readlink -f` works. Only if it does you set the `rlink` variable to something. Otherwise `rlink` is empty.
Later you check whether `rlink` is empty and if so you exist. Why not directly exiting if the first check failed?

Reply:
> Why not directly exiting if the first check failed?

Because it is ok to not have `readlink -f` support, as long as `SOURCE_PATH` is set. There are 3 scenarios:

1. `readlink -f` works - current, with either `SOURCE_PATH` set or not
2. `readlink -f` does not work and `SOURCE_PATH` set: ok, but warn
3. `readlink -f` does not work and `SOURCE_PATH` not set: exit

Reply:
Alright, thanks!